src/cnxml/jsx.ts

```diff
@@ -29,8 +30,28 @@ export interface Element {
 
 export type Attributes = Omit<JSX.IntrinsicAttributes, 'children'> & { [key: string]: unknown }
 
+/** A processing instruction */
+export interface ProcessingInstruction {
+    /** PI's target */
+    target: string
+    /** PI's value */
+    value: string
+}
+
 /** Any value that can be used as child of a JSX element */
-export type Node = Element | globalThis.Node | string | Node[] | null
+export type Node = Element | ProcessingInstruction | globalThis.Node | string | Node[] | null
+
+export const Node = {
+    isElement(node: Node): node is Element {
+        return Boolean(node
+            && !Array.isArray(node)
+            && !Node.isProcessingInstruction(node)
+            && !(node instanceof globalThis.Node))
+    },
+    isProcessingInstruction(node: Node): node is ProcessingInstruction {
+        return Boolean(node && typeof (node as ProcessingInstruction).target === 'string')
```

Comment:
Please don't use `Boolean(expression)` to check for truthness. If you want to check that `node` is not `null` do so with `node != null`.

`isElement` should check that `node` is an `Element` directly, rather than checking that it is _not_ any of the other possible types.

Reply:
About the check for `isElement` - do you want me to check if the `node` is an object that contains `name` and that the name is an object that contains local of type string and that the node contains `attributes` of type object and also `children` that are of type `node` (which will require creating `Node.isNode` function)?

Reply:
I don't think we need a full check, after all you are taking `node: Node`. I think something equivalent to `isPlainObject(node) && typeof node.name === 'object'` would suffice (using `isPlainObject` from package `is-plain-object`, it's already in our dependency tree as Slate uses it).